nextest-metadata/src/test_list.rs

```diff
@@ -510,6 +516,45 @@ pub struct RustNonTestBinarySummary {
     pub path: Utf8PathBuf,
 }
 
+/// Serialized representation of the host platform.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct HostPlatformSummary {
+    /// The host platform, if specified.
+    pub platform: PlatformSummary,
+
+    /// The libdir for the host platform.
+    ///
+    /// Empty if failed to discover.
+    #[serde(default)]
+    pub libdir: Option<Utf8PathBuf>,
+}
+
+/// Serialized representation of the target platform.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct TargetPlatformSummary {
+    /// The target platform, if specified.
+    pub platform: PlatformSummary,
+
+    /// The libdir for the target platform.
+    ///
+    /// Empty if failed to discover.
+    #[serde(default)]
+    pub libdir: Option<Utf8PathBuf>,
+}
+
+/// Serialized representation of the host and the target platform.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct BuildPlatformsSummary {
+    /// The target platform used while compiling the Rust artifacts.
+    pub host: HostPlatformSummary,
+
+    /// The host platform used while compiling the Rust artifacts.
+    pub targets: Vec<TargetPlatformSummary>,
```

Comment:
```suggestion
    /// The host platform used while compiling the Rust artifacts.
    pub host: HostPlatformSummary,

    /// The target platforms used while compiling the Rust artifacts.
    ///
    /// With current versions of nextest, this will contain at most one element.
    pub targets: Vec<TargetPlatformSummary>,
```